utilities/rate_limiters/write_amp_based_rate_limiter.cc

```diff
@@ -77,13 +77,33 @@ WriteAmpBasedRateLimiter::~WriteAmpBasedRateLimiter() {
 
 void WriteAmpBasedRateLimiter::SetBytesPerSecond(int64_t bytes_per_second) {
   assert(bytes_per_second > 0);
-  if (auto_tuned_) {
+  if (auto_tuned_.load(std::memory_order_relaxed)) {
     max_bytes_per_sec_.store(bytes_per_second, std::memory_order_relaxed);
   } else {
     SetActualBytesPerSecond(bytes_per_second);
   }
 }
 
+void WriteAmpBasedRateLimiter::SetAutoTuned(bool auto_tuned) {
+  MutexLock g(&auto_tuned_mutex_);
+  if (auto_tuned_.load(std::memory_order_relaxed) != auto_tuned) {
```

Comment:
Let's use seq_cst order for all access to `auto_tuned_`?

Reply:
I think seq_cst is equivalent to acq_rel if only one variable is tagged. I went through the logic again and added some tags.

Reply:
yeah, using acquire/release ordering is even better, but sometimes they are hard to reason about. But as we are considering use arm, it make more sense for us to take some effort and use the acquire/release semantics if suitable.